### PR TITLE
[Diffusion] Feat: support teacache for glm-image, qwen-image et.al

### PR DESCRIPTION
**Pull request by RuixiangMa**
## Accuracy Tests

- All tests used default parameters，my GPU memory's tight, so I enabled layer offload for the bigger model（Qwen-image  et.al）
- Results are directly comparable under identical conditions.

| Model | ❌ No TeaCache | ✅ TeaCache  |
| :---: | :---: | :---: |
| **Qwen-Image-2512** | **156.12**<br>a_young_girl_with_flowing_long_hair_wearing_a_white_halter_dress_and_smiling_sweetly _The_backgroun_20260312-154020_5e39b8af | **61.29**<br>a_young_girl_with_flowing_long_hair_wearing_a_white_halter_dress_and_smiling_sweetly _The_backgroun_20260312-154538_8358cc54 |
| **GLM-IMAGE** | **87.87**<br>a_young_girl_with_flowing_long_hair_wearing_a_white_halter_dress_and_smiling_sweetly _The_backgroun_20260312-160354_cc167b0e | **69.83**<br>a_young_girl_with_flowing_long_hair_wearing_a_white_halter_dress_and_smiling_sweetly _The_backgroun_20260312-160614_07782f69|
| **flux1** | **61.87**<br>a_young_girl_with_flowing_long_hair_wearing_a_white_halter_dress_and_smiling_sweetly _The_backgroun_20260312-163913_254194f4 | **21.98**<br>a_young_girl_with_flowing_long_hair_wearing_a_white_halter_dress_and_smiling_sweetly _The_backgroun_20260312-164154_e8906659 |
| **flux2** | **5.95**<br>a_young_girl_with_flowing_long_hair_wearing_a_white_halter_dress_and_smiling_sweetly _The_backgroun_20260312-164428_baf3c7b9 | **5.50**<br>a_young_girl_with_flowing_long_hair_wearing_a_white_halter_dress_and_smiling_sweetly _The_backgroun_20260312-164526_3dab1de2|

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/models/dits/flux.py [hunk 1]
```diff
@@ -865,6 +866,12 @@ def forward(
                 [diffusers.models.attention_processor](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_processor.py).
 
         """
+        forward_context = get_forward_context()
+        forward_batch = forward_context.forward_batch if forward_context else None
+        self.enable_teacache = forward_batch is not None and getattr(
+            forward_batch, "enable_teacache", False
+        )
```
[the remark this answers is not in the document]
Comment (RuixiangMa) [reply python/sglang/multimodal_gen/runtime/models/dits/flux.py]: The current explicit approach, while duplicated, is clearer and less error-prone.

File: python/sglang/multimodal_gen/runtime/models/dits/glm_image.py [hunk 2]
```diff
@@ -848,16 +855,27 @@ def forward(
...
-            hidden_states, encoder_hidden_states = block(
-                hidden_states,
-                encoder_hidden_states,
-                temb,
-                image_rotary_emb,
-                attention_mask,
-                attention_kwargs,
-                kv_cache=kv_caches[idx] if kv_caches is not None else None,
-            )
+        should_skip_forward = self.should_skip_forward_for_cached_states(temb=temb)
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/models/dits/glm_image.py]: we need a helper function to simplify this logic

Comment (RuixiangMa) [reply python/sglang/multimodal_gen/runtime/models/dits/glm_image.py]: Done, pls have a look
